lib/puppet/util/run_mode.rb

```diff
@@ -38,6 +38,10 @@ def log_dir
         "$vardir/log"
       end
 
+      def conf_dir
+        "$puppetdir/config"
+      end
+
```

Comment:
@justinstoller @ahpook - Here is the answer to `$puppetdir` not expanding - turns out @roidelapluie already solved it for us.